### ISSUE TITLE
Find rich text component for JavaFX

### ISSUE THREAD

**Issue by ManfredKarrer** [bisq-network/bisq#1059]
Find rich text component for JavaFX to support inline links (no Webkit as it it too heavy) https://github.com/TomasMikula/ might have/know something.

Main use case is in popups where we want to have better formatted text with hyperlink support.

**Comment by ghost** [comment]
For which use-case? JavaFX isn't so popular, I found only one candidate - https://github.com/FXMisc/RichTextFX

**Comment by ManfredKarrer** [comment]
Yes that one I tried in the early days as well but could not get it working for my use cases.
It is to display rich text (hyperlinks, formatted text) in the popups and other text. Atm we use plain TextFields/Labels or custom components for links. Beside low level Text API (too much effort) and WebKit (to resource heavy) I did not found any thing for doing the job well.
Tomas is at least a very good UI dev, so even if his component does not fulfill our needs maybe he knows something or can build it if it really does not exist. Markdown format support would be good as well.

I will edit above text to make use case more clear.

**Comment by ghost** [comment]
Yes, as an alternative it can be `TextFlow` standard control from JavaFX, and I found also a good comparison between existing alternative - https://www.namekdev.net/2016/03/javafx-comparison-of-rich-text-components/

> In Java 8 you can you can use the TextFlow to style your text rather than a combination of different Text instances in a FlowPane.

**Comment by ghost** [comment]
Ex. 
![image](https://user-images.githubusercontent.com/39793/33597835-e293a47c-d9a0-11e7-84c0-f68ec66965f9.png)

**Comment by ManfredKarrer** [comment]
Yes with TextFlow I worked a bit but was a lot of effort. 
Another reason why I don't want to add WebView (WebKit) is because of security reasons.
Your example is with RichTextFX?
Can you try to apply that to a popup in Bisq with some basic formatting changes (bold, hyperlinks)? Does it support inline images? Not a hard requirement but would be nice to have.
The text comes from the properties files and the format code need to work from there and should be still readable (translations done by non devs).

**Comment by ghost** [comment]
It was example from `TextFlow` component. Yea I can try, but not immediately. First of all I want to set up dev env (regtest).

**Comment by ManfredKarrer** [comment]
We need to have the format in the text like markdown or html. Not sure if that works with TextFlow without build a own parsing framework.

**Comment by ManfredKarrer** [comment]
Here is an example for a popup text:

payment.clearXchange.info=Please be sure that you fulfill the requirements for the usage of Zelle (ClearXchange).\n\n\
1. You need to have your Zelle (ClearXchange) account verified on their platform \
before starting a trade or creating an offer.\n\n\
2. You need to have a bank account at one of the following member banks:\n\
\t● Bank of America\n\
\t● Capital One P2P Payments\n\
\t● Chase QuickPay\n\
\t● FirstBank Person to Person Transfers\n\
\t● Frost Send Money\n\
\t● U.S. Bank Send Money\n\
\t● Wells Fargo SurePay\n\n\
3. You need to be sure to not exceed the daily or monthly Zelle (ClearXchange) transfer limits.\n\n\
Please use Zelle (ClearXchange) only if you fulfill all those requirements, \
otherwise it is very likely that the Zelle (ClearXchange) transfer fails and the trade ends up in a dispute.\n\
If you have not fulfilled the above requirements you will lose your security deposit.\n\n\
  Please also be aware of a higher chargeback risk when using Zelle (ClearXchange).\n\
  For the {0} seller it is highly recommended \
to get in contact with the {1} buyer by using the provided email address or mobile number to verify that he or she \
  is really the owner of the Zelle (ClearXchange) account.

**Comment by ManfredKarrer** [comment]
Looks like: 
<img width="930" alt="screen shot 2017-12-05 at 07 58 31" src="https://user-images.githubusercontent.com/1449498/33608296-1f33675a-d992-11e7-8e59-05e705fae880.png">

**Comment by ghost** [comment]
And that a problem with current impl., what we should to change or what a new requirements? Speed? Images? Hyperlinks? Internal links? All of them? ;)

**Comment by ManfredKarrer** [comment]
Btw. I think inline image support IS a requirement, otherwise we cannot "beautify" those popups enough. @ripcurlx What do you think?

**Comment by ManfredKarrer** [comment]
Speed is not a requirement. But yes inline images, Hyperlinks are. Internal links (handlers) would be good as well.

**Comment by ripcurlx** [comment]
The problem with the current popups is, that we present important content displaying text without lots of formatting. As we know that people hardly read text we need to present it in a more visual way so important information is easy to digest. This is especially important as we get more non-tech-savvy people on the platform. I haven't started creating example layouts, how I want the popups to be, so I'll start with that first, to give you an idea on the requirements.

**Comment by ManfredKarrer** [comment]
Yes great. If we have an example it is easier to see which features are really needed. A bit of formatting is even possible with current version but I assume you might want to use images as well?

**Comment by ripcurlx** [comment]
As we'll use custom layout for every important popup, there's no big benefit doing it with a Rich Text component, that's limiting the layout in some way.
